src/riak_kv_index_hashtree.erl

```diff
@@ -564,7 +598,13 @@ do_new_tree(Id, State=#state{trees=Trees, path=Path}) ->
     Trees2 = orddict:store(Id, NewTree, Trees),
     State#state{trees=Trees2}.
 
--spec do_get_lock(any(), pid(), state()) -> {not_built | ok | already_locked, state()}.
+-spec do_get_lock(any(), pid(), state()) -> {building | not_built | ok | already_locked, state()}.
+ do_get_lock(_, _, State) when State#state.waiting_for_lock /= undefined ->
+    lager:debug("Already waiting for lock: ~p ~p", [State#state.index, State#state.waiting_for_lock]),
+    {already_locked, State};
+do_get_lock(_, _, State) when is_pid(State#state.built) ->
+    lager:debug("Building: ~p :: ~p", [State#state.index, State#state.built]),
+    {building, State};
```

Comment:
Is it really necessary to introduce the `building` reply instead of returning not_built? Notice that this makes the get_lock call also start returning this, which propagates down many paths in the Riak code.